android/src/main/java/com/attentivereactnativesdk/AttentiveReactNativeSdkModule.java

```diff
@@ -244,4 +307,257 @@ private List<Item> buildItems(ReadableArray rawItems) {
+
+    // Tab click listeners
+    currentTab.setOnClickListener(v -> {
+      updateCurrentEventContent(contentText, currentEvent, currentData);
+      currentTab.setEnabled(false);
+      historyTab.setEnabled(true);
+    });
+
+    historyTab.setOnClickListener(v -> {
+      updateHistoryContent(contentText);
+      currentTab.setEnabled(true);
+      historyTab.setEnabled(false);
+    });
+
+    // Set initial tab state
+    currentTab.setEnabled(false);
+    historyTab.setEnabled(true);
+
+    // Setup close button to remove the overlay from the root view
+    closeButton.setOnClickListener(v -> {
+      rootView.removeView(mainLayout);
+    });
+
+    // Add the overlay directly to the activity's root view
+    rootView.addView(mainLayout);
+
+    // Auto-dismiss after 8 seconds (longer for history viewing)
+    new android.os.Handler(android.os.Looper.getMainLooper()).postDelayed(() -> {
+      // Check if the view is still attached before removing
+      if (mainLayout.getParent() != null) {
+        rootView.removeView(mainLayout);
+      }
+    }, 8000);
+  }
+
+  private void updateCurrentEventContent(android.widget.TextView contentText, String event, Map<String, Object> data) {
+    StringBuilder content = new StringBuilder();
+    content.append("Event: ").append(event).append("\n\n");
+
+    try {
+      org.json.JSONObject jsonObject = new org.json.JSONObject(data);
+      content.append(jsonObject.toString(2));
+    } catch (Exception e) {
+      content.append("Data: ").append(data.toString());
+    }
+
+    contentText.setText(content.toString());
+  }
+
+  private void updateHistoryContent(android.widget.TextView contentText) {
+    if (debugHistory.isEmpty()) {
+      contentText.setText("No events recorded in this session yet.");
+      return;
+    }
+
+    StringBuilder content = new StringBuilder();
+    content.append("Session History (").append(debugHistory.size()).append(" events):\n\n");
+
+    // Show events in reverse order (newest first)
+    for (int i = debugHistory.size() - 1; i >= 0; i--) {
+      DebugEvent event = debugHistory.get(i);
+      content.append("───────────────────────────────\n");
+      content.append("[").append(event.getFormattedTime()).append("] ");
+      content.append(event.getEventType()).append("\n");
+
+      // Add summary
+      String summary = event.getSummary();
+      if (!summary.isEmpty()) {
+        content.append("Summary: ").append(summary).append("\n");
+      }
+
+      content.append("\nPayload:\n");
+      try {
+        org.json.JSONObject jsonObject = new org.json.JSONObject(event.getData());
+        content.append(jsonObject.toString(2));
+      } catch (Exception e) {
+        content.append(event.getData().toString());
+      }
+      content.append("\n\n");
+    }
+
+    contentText.setText(content.toString());
+  }
+
+  // Debug Event class for session history
+  private static class DebugEvent {
```

Comment:
Could you make this a kotlin class? Let's try to make all new android code use kotlin instead of java.

Reply:
It would be great if you could convert this whole file to kotlin, is that feasible?